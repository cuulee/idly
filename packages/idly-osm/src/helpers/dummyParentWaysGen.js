"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var immutable_1 = require("immutable");
function dummyParentWaysGen(obj) {
    return immutable_1.Map(obj);
}
exports.dummyParentWaysGen = dummyParentWaysGen;
