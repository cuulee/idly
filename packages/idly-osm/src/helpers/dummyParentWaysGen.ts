import { ParentWays } from 'idly-common/lib/osm/structures';

export function dummyParentWaysGen(obj: any) {
  const parentWays: ParentWays = new Map();
  Object.keys(obj).forEach(k => parentWays.set(k, obj[k]));
  return parentWays;
}
