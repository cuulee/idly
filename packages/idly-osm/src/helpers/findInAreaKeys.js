"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var areaKeys_1 = require("idly-data/lib/areaKeys/areaKeys");
// `highway` and `railway` are typically linear features, but there
// are a few exceptions that should be treated as areas, even in the
// absence of a proper `area=yes` or `areaKeys` tag.. see #4194
var lineKeys = {
    highway: {
        rest_area: true,
        services: true
    },
    railway: {
        roundhouse: true,
        station: true,
        traverser: true,
        turntable: true,
        wash: true
    }
};
function findInAreaKeys(tags) {
    var areaKeys = areaKeys_1.getAreaKeys();
    for (var key in tags) {
        if (key in areaKeys && !(tags[key] in areaKeys[key])) {
            return true;
        }
        if (key in lineKeys && tags[key] in lineKeys[key]) {
            return true;
        }
    }
    return false;
}
exports.findInAreaKeys = findInAreaKeys;
