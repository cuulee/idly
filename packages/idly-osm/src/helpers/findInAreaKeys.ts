import { Tags } from 'idly-common/lib/osm/structures';
import { getAreaKeys } from 'idly-data/lib/areaKeys/areaKeys';

// `highway` and `railway` are typically linear features, but there
// are a few exceptions that should be treated as areas, even in the
// absence of a proper `area=yes` or `areaKeys` tag.. see #4194
const lineKeys: any = {
  highway: {
    rest_area: true,
    services: true
  },
  railway: {
    roundhouse: true,
    station: true,
    traverser: true,
    turntable: true,
    wash: true
  }
};

export function findInAreaKeys(tags: Tags) {
  const areaKeys = getAreaKeys();
  for (const key in tags) {
    if (key in areaKeys && !(tags[key] in areaKeys[key])) {
      return true;
    }
    if (key in lineKeys && tags[key] in lineKeys[key]) {
      return true;
    }
  }
  return false;
}
