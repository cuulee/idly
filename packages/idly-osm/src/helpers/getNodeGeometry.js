"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var structures_1 = require("idly-common/lib/osm/structures");
function getNodeGeometry(id, parentWay) {
    if (parentWay.size === 0)
        return structures_1.OsmGeometry.POINT;
    return structures_1.OsmGeometry.VERTEX;
}
exports.getNodeGeometry = getNodeGeometry;
