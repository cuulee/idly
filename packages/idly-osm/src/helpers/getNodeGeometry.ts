import {
  EntityId,
  NodeGeometry,
  OsmGeometry,
  ParentWays
} from 'idly-common/lib/osm/structures';
import { ImSet } from 'idly-common/lib/misc/immutable';

export function getNodeGeometry(
  id: EntityId,
  parentWay: ImSet<string>
): NodeGeometry {
  if (parentWay.size === 0) return OsmGeometry.POINT;

  return OsmGeometry.VERTEX;
}
