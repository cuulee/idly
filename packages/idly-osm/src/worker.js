"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var immutable_1 = require("immutable");
var structures_1 = require("idly-common/lib/osm/structures");
var config_1 = require("./config/config");
var node_1 = require("./parsers/node");
var way_1 = require("./parsers/way");
function nameSpaceKeys(name, obj) {
    var newObj = {};
    Object.keys(obj).forEach(function (k) {
        newObj[name + '--' + k] = obj[k];
    });
    return newObj;
}
/**
 * meant to run purely on a separate thread.
 * @param entityTable
 * @param parentWays
 */
function onParseEntities(entityTable, parentWays) {
    console.log('onParseEntities called worker !!!');
    var fProps = new Map();
    entityTable.forEach(function (entity, id) {
        if (entity.type === structures_1.EntityType.NODE) {
            // @TOFIX why do we need to send the entire parentWays lol.
            var x = nameSpaceKeys(config_1.PLUGIN_NAME, node_1.nodePropertiesGen(entity, parentWays.get(entity.id) || immutable_1.Set()));
            fProps.set(id, x);
        }
        if (entity.type === structures_1.EntityType.WAY) {
            fProps.set(id, nameSpaceKeys(config_1.PLUGIN_NAME, way_1.wayPropertiesGen(entity)));
        }
    });
    return fProps;
}
exports.onParseEntities = onParseEntities;
