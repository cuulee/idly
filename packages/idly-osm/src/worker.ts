import { ImSet } from 'idly-common/lib/misc/immutable';
import { EntityTable, EntityType, FeaturePropsTable, ParentWays } from 'idly-common/lib/osm/structures';

import { PLUGIN_NAME } from './config/config';
import { nodePropertiesGen } from './parsers/node';
import { wayPropertiesGen } from './parsers/way';

function nameSpaceKeys(name, obj: { [index: string]: string }) {
  var newObj = {};
  Object.keys(obj).forEach(k => {
    newObj[name + '--' + k] = obj[k];
  });
  return newObj;
}
/**
 * meant to run purely on a separate thread.
 * @param entityTable 
 * @param parentWays 
 */
export function onParseEntities(
  entityTable: EntityTable,
  parentWays: ParentWays
): FeaturePropsTable {
  console.log('onParseEntities called worker !!!');
  const fProps: FeaturePropsTable = new Map();

  entityTable.forEach((entity, id) => {
    if (entity.type === EntityType.NODE) {
      // @TOFIX why do we need to send the entire parentWays lol.
      var x = nameSpaceKeys(
        PLUGIN_NAME,
        nodePropertiesGen(entity, parentWays.get(entity.id) || ImSet())
      );
      fProps.set(id, x);
    }
    if (entity.type === EntityType.WAY) {
      fProps.set(id, nameSpaceKeys(PLUGIN_NAME, wayPropertiesGen(entity)));
    }
  });
  return fProps;
}
