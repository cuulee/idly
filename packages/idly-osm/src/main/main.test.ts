import { nodeFactory } from 'idly-common/lib/osm/nodeFactory';
import { genLngLat } from 'idly-common/lib/osm/genLngLat';
import { tagsFactory } from 'idly-common/lib/osm/tagsFactory';
import { wayFactory } from 'idly-common/lib/osm/wayFactory';
import { entityTableGen } from 'idly-common/lib/osm/entityTableGen';

import { onParseEntities } from '../worker';
import { dummyParentWaysGen } from '../helpers/dummyParentWaysGen';
import { Set as ImSet } from 'immutable';

describe('test IdlyOSM', () => {
  describe('onParseEntities', () => {
    const n1 = nodeFactory({
      id: 'n1',
      loc: genLngLat({ lon: 15, lat: 10 }),
      tags: tagsFactory({ k: 'k' })
    });
    const n2 = nodeFactory({
      id: 'n2',
      loc: genLngLat({ lon: 5, lat: 12 }),
      tags: tagsFactory({ k: 'k' })
    });
    const n3 = nodeFactory({
      id: 'n3',
      loc: genLngLat({ lon: 15, lat: 10 }),
      tags: tagsFactory({ k: 'k' })
    });
    const n4 = nodeFactory({
      id: 'n4',
      loc: genLngLat({ lon: 0, lat: 10 }),
      tags: tagsFactory({ k: 'k' })
    });

    const w1 = wayFactory({
      id: 'w1',
      nodes: ['n1', 'n2', 'n4'],
      tags: tagsFactory({ highway: 'residential' })
    });

    const w2 = wayFactory({
      id: 'w2',
      nodes: ['n1', 'n3'],
      tags: tagsFactory({ highway: 'residential' })
    });
    var table = entityTableGen([n1, n2, n3, n4, w1, w2]);
    var parentWays = dummyParentWaysGen({
      n1: ImSet(['w1', 'w2']),
      n2: ImSet(['w1']),
      n3: ImSet(['w2']),
      n4: ImSet(['w1'])
    });
    it('should name space properties', () => {
      var propsAdded = onParseEntities(table, parentWays);
      expect(propsAdded).toMatchSnapshot();
    });
  });
});
