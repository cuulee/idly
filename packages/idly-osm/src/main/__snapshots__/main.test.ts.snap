// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test IdlyOSM onParseEntities should name space properties 1`] = `
Map {
  "n1" => Object {
    "osm_basic--geometry": "vertex",
    "osm_basic--icon": "circle",
    "osm_basic--name": undefined,
  },
  "n2" => Object {
    "osm_basic--geometry": "vertex",
    "osm_basic--icon": "circle",
    "osm_basic--name": undefined,
  },
  "n3" => Object {
    "osm_basic--geometry": "vertex",
    "osm_basic--icon": "circle",
    "osm_basic--name": undefined,
  },
  "n4" => Object {
    "osm_basic--geometry": "vertex",
    "osm_basic--icon": "circle",
    "osm_basic--name": undefined,
  },
  "w1" => Object {
    "osm_basic--geometry": "line",
    "osm_basic--icon": "highway-residential",
    "osm_basic--name": undefined,
    "osm_basic--tagsClass": "tag-highway",
    "osm_basic--tagsClassType": "tag-highway-residential",
  },
  "w2" => Object {
    "osm_basic--geometry": "line",
    "osm_basic--icon": "highway-residential",
    "osm_basic--name": undefined,
    "osm_basic--tagsClass": "tag-highway",
    "osm_basic--tagsClassType": "tag-highway-residential",
  },
}
`;
